manifests/init.pp

```diff
@@ -451,7 +451,7 @@
 
   # Validate dns params
   validate_bool($dns)
-  validate_string($dns_interface, $dns_provider, $dns_reverse, $dns_server, $keyfile)
```

Comment:
Validation shouldn't be removed - if you're trying to change the data type then you need to add validation for the new type (e.g. validate_array), and update default values in params.pp too. Even consider supporting both strings and arrays, as it's an incompatible change.